xbmc/platform/win32/storage/discs/Win32DiscDriveHandler.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ *  Copyright (C) 2022 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "Win32DiscDriveHandler.h"
+
+#include "utils/log.h"
+
+#include "platform/win32/WIN32Util.h"
+
+#include <memory>
+
+std::shared_ptr<IDiscDriveHandler> IDiscDriveHandler::CreateInstance()
+{
+  return std::make_shared<CWin32DiscDriveHandler>();
+}
+
+DriveState CWin32DiscDriveHandler::GetDriveState(const std::string& devicePath)
+{
+  DriveState driveState = DriveState::DRIVE_NOT_READY;
+  auto status = static_cast<CWin32DiscDriveState>(CWIN32Util::GetDriveStatus(devicePath));
```

Comment:
This isn't really a great use of an `enum class` You should probably just return the enum from `CWIN32Util::GetDriveStatus` rather than trying to map it here.

Reply:
In the past it was just using int values in the switch statement. However in some other PR you requested some sort of mapping to avoid the integer values.
I am a bit lost of how to best handle this.  I agree the casting isnt great but in this particular case we can only have 0 (closed), 1 (open) or an error (negative values).
Returning the enum is not an option as the goal of DriveState is to abstract this across platforms.

Reply:
Maybe I can just define those ints in the anonymous namespace of the file as constexpr. That way we can avoid the cast but still have something meaningful in the switch case.

Reply:
Something like this is fine.
```c++
std::map<CWin32DiscDriveState, DriveState> DriveStateMap =
{
  {CWin32DiscDriveState::DRIVE_ERROR, DriveState::DRIVE_NOT_READY},
  {CWin32DiscDriveState::NO_MEDIA, DriveState:: DRIVE_CLOSED_NO_MEDIA},
  ...
};
```